sentry-graphql-java/src/main/java/io/sentry/graphql/SentryInstrumentation.java

```diff
@@ -0,0 +1,167 @@
+
+public final class SentryInstrumentation extends SimpleInstrumentation {
+  private final @NotNull IHub hub;
+  private final @Nullable BeforeSpanCallback beforeSpan;
+
+  public SentryInstrumentation(
+      final @NotNull IHub hub, final @Nullable BeforeSpanCallback beforeSpan) {
+    this.hub = Objects.requireNonNull(hub, "hub is required");
+    this.beforeSpan = beforeSpan;
+  }
+
+  public SentryInstrumentation(final @NotNull IHub hub) {
+    this(hub, null);
+  }
+
+  public SentryInstrumentation() {
+    this(HubAdapter.getInstance());
+  }
+
+  @Override
+  public @NotNull InstrumentationState createState() {
+    return new TracingState();
+  }
+
+  @Override
+  public @NotNull InstrumentationContext<ExecutionResult> beginExecution(
+      final @NotNull InstrumentationExecutionParameters parameters) {
+    final TracingState tracingState = parameters.getInstrumentationState();
+    tracingState.setTransaction(hub.getSpan());
+    return super.beginExecution(parameters);
+  }
+
+  @Override
+  @SuppressWarnings("FutureReturnValueIgnored")
+  public @NotNull DataFetcher<?> instrumentDataFetcher(
+      final @NotNull DataFetcher<?> dataFetcher,
+      final @NotNull InstrumentationFieldFetchParameters parameters) {
+    // We only care about user code
+    if (parameters.isTrivialDataFetcher()) {
+      return dataFetcher;
+    }
+
+    return environment -> {
+      final TracingState tracingState = parameters.getInstrumentationState();
+      final ISpan transaction = tracingState.getTransaction();
+      if (transaction != null) {
+        final ISpan span = transaction.startChild(findDataFetcherTag(parameters));
+        try {
+          final Object result = dataFetcher.get(environment);
+          if (result instanceof CompletableFuture) {
+            ((CompletableFuture<?>) result)
+                .whenComplete(
+                    (r, ex) -> {
+                      if (ex != null) {
+                        span.setThrowable(ex);
+                        span.setStatus(SpanStatus.INTERNAL_ERROR);
+                      }
+                      finish(span, environment, r);
+                    });
+          } else {
+            finish(span, environment, result);
+          }
+          return result;
+        } catch (Exception e) {
+          span.setThrowable(e);
+          span.setStatus(SpanStatus.INTERNAL_ERROR);
+          finish(span, environment);
+          throw e;
+        }
+      } else {
+        return dataFetcher.get(environment);
+      }
+    };
+  }
+
+  private void finish(
+      final @NotNull ISpan span,
+      final @NotNull DataFetchingEnvironment environment,
+      final @Nullable Object result) {
+    if (beforeSpan != null) {
+      final ISpan newSpan = beforeSpan.execute(span, environment, result);
+      if (newSpan == null) {
+        // span is dropped
+        span.getSpanContext().setSampled(false);
+      } else {
+        newSpan.finish();
+      }
+    } else {
+      span.finish();
```

Comment:
Thats something we agreed on, that by default in successful scenario we don't pass status. It's the same in all integrations.

Reply:
so the difference between this one and the other integrations is that the other integrations always try to get the status from the HTTP response code, but here we always leave it without status.
can we do the same here in case the request is successful?

Reply:
it is, https://github.com/getsentry/develop/issues/307
there was the discussion but none of us worked on the docs yet, PR welcomed :P